bundles/org.openhab.ui/web/src/components/item/metadata/item-metadata-alexa.vue

```diff
@@ -47,43 +77,125 @@ export default {
-      return [...this.classesDefs].sort((a, b) => {
-        return a.localeCompare(b)
-      })
+      return [...this.classesDefs]
+        .filter((cl) => this.supportsItemType(cl) && this.supportsGroupType(cl) && !this.requiresGroupAttributes(cl))
+        .sort((a, b) => a.localeCompare(b))
+    },
+    defaultClasses () {
+      return this.orderedClasses.filter((cl) => cl.split('.').length === 1)
+    },
+    genericClasses () {
+      return this.orderedClasses.filter((cl) => cl.split('.').length === 2 && this.supportsMultiInstance(cl))
+    },
+    specificClasses () {
+      return this.orderedClasses.filter((cl) => cl.split('.').length === 2 && !this.supportsMultiInstance(cl))
     },
     parameters () {
-      if (!this.classes) return []
-      if (!this.multiple) {
-        return AlexaDefinitions['label:' + this.classes] || AlexaDefinitions['endpoint:' + this.classes] || [...AlexaDefinitions[this.classes]]
-      }
-      const params = []
-      this.classes.forEach((c) => {
-        for (const p of AlexaDefinitions[c]) {
-          if (!params.find(p2 => p2.name === p.name)) params.push(p)
+      return this.classes.reduce((parameters, cl) => {
+        const def = AlexaDefinitions[cl]
+        const params = def ? this.itemType === 'Group' ? def.groupParameters : def.parameters : []
+        for (const p of params.map((p) => p(this.item, this.metadata.config)).flat()) {
+          if (!parameters.find((e) => e.name === p.name)) parameters.push(p)
         }
-      })
-      return params
+        return parameters
+      }, [])
+    },
+    groupCapabilities () {
+      return this.item.members
+        .filter((mbr) => mbr.metadata && mbr.metadata.alexa)
+        .reduce((caps, mbr, idx, arr) => caps.concat(
+          mbr.metadata.alexa.value.split(',').map((cl) => ({
+            name: cl.split('.').pop().trim() || 'N/A',
+            item: mbr.name,
+            isIgnored:
+              !this.isSupportedGroupAttribute(cl) ||
+              !this.hasRequiredGroupAttributes(cl, arr) ||
+              (!this.supportsMultiInstance(cl) &&
+                arr.findIndex((mbr) => mbr.metadata.alexa.value.split(',').includes(cl)) !== idx)
+          }))
+        ), [])
+    },
+    groupsDescription () {
+      return this.item.groups.map((g) => `${g.label || g.name} (${g.metadata.alexa.value})`).join(', ')
+    },
+    link () {
+      return 'https://www.openhab.org/link/alexa#'.concat(
+        this.itemType === 'Group'
+          ? 'group-endpoint'
+          : this.classes.length === 0 || !this.classesDefs.includes(this.classes[0])
+            ? this.item.groups.length
+              ? 'group-endpoint'
+              : 'single-endpoint'
+            : this.classes[0].indexOf('.') === -1
+              ? this.classes[0] === 'Scene' || this.classes[0] === 'Activity'
+                ? this.classes[0].toLowerCase()
+                : 'device-types'
+              : this.classes[0].split('.')[1].toLowerCase()
```

Comment:
This could surely be made more readable with some regular "if" statements :)

Reply:
Done including the different documentation link support as mentioned above.